src/Manager.cpp

```diff
@@ -142,5 +142,26 @@ namespace ClassProject
 
     }
 
+    /**
+    * @brief getTopVarName
+    *
+    * returns the name of the top variable of the given node.
+    * @param root: ID of the node.
+    * @return returns the name of the top variable.
+    */
+    std::string Manager::getTopVarName(const BDD_ID &root) {
+        BDD_ID temp;
+        for (auto &it: Table) {
+            if (it.second.id == root ) {
+                temp = it.first.TopVar;
+            }
+            if (it.second.id == temp) {
+                return it.second.label;
+            }
+        }
+    }
```

Comment:
I think, there's a problem with this function and I'm quite surprised it passed the tests.
First, you're looking for the TopVar, which is okay, but in the same loop, you're trying to find the TopVar label. You function is working only in the case the iterator find the root ID **first**.
The problem may happen when the TopVar ID is first in the map, before you find your root ID. Then the function may never find TopVar ID  (`it.second.id == temp` condition). Besides that, its never safe to check `it.second.id == temp` when `temp` is not ititialized.
I would recommend to use `TopVar()` function to find TopVar ID first, and then find its label using loop like this.

Reply:
Resolved.